server/src/main/java/org/eclipse/lsp/cobol/core/model/tree/ParagraphDefinitionNameNode.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2022 Broadcom.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *    Broadcom, Inc. - initial API and implementation
+ *
+ */
+package org.eclipse.lsp.cobol.core.model.tree;
+
+import com.google.common.collect.ImmutableList;
+import lombok.Getter;
+import org.eclipse.lsp.cobol.core.model.Locality;
+import org.eclipse.lsp.cobol.core.model.SyntaxError;
+import org.eclipse.lsp4j.Location;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/** The class represents paragraphs name node in COBOL grammar. */
+@Getter
+public class ParagraphDefinitionNameNode extends Node implements Context {
+  private final String paragraphName;
+  List<Location> definitions = new ArrayList<>();
+  List<Location> usageLocations = new ArrayList<>();
```

Comment:
Now I see what happens.
You collect definitions and usages in that node. But you can have several `ParagraphDefinitionNameNode` just because the user can define paragraphs twice.
What is it about having a class with only a list of definitions and usages? That class can be used inside `ProgramNode` only. And you can have only one instance of that class for each unique paragraph because it is not related to the user input.